org.jacoco.core.test.validation.java21/src/org/jacoco/core/test/validation/java21/RecordPatternsTest.java

```diff
@@ -26,4 +26,12 @@ public RecordPatternsTest() {
 		super(RecordPatternsTarget.class);
 	}
 
+	public void assertSwitchStatementLastCase(final Line line) {
+		if (isJDKCompiler) {
+			assertFullyCovered(line);
+		} else {
+			assertPartlyCovered(line);
```

Comment:
To me seems that this is related to https://github.com/eclipse-jdt/eclipse.jdt.core/issues/773